htdocs/core/class/commoninvoice.class.php

```diff
@@ -1186,126 +1194,11 @@ public function makeStripeSepaRequest($fuser, $did = 0, $type = 'direct-debit',
 											// Default description used for label of event. Will be overwrite by another value later.
 											$description = 'Stripe payment request OK (' . $charge->id . ') from makeStripeSepaRequest: ' . $FULLTAG;
 
-
-											// @TODO LMR Save request to status pending instead of done. Done should be set with a webhook.
-
-
-											$db = $this->db;
-
-											$ipaddress = getUserRemoteIP();
-
-											$TRANSACTIONID = $charge->id;
-											$currency = $conf->currency;
-											$paymentmethod = 'stripe';
-											$emetteur_name = $charge->customer;
-
-											// Same code than into paymentok.php...
-
-											$paymentTypeId = 0;
-											if ($paymentmethod == 'paybox') {
-												$paymentTypeId = $conf->global->PAYBOX_PAYMENT_MODE_FOR_PAYMENTS;
-											}
-											if ($paymentmethod == 'paypal') {
-												$paymentTypeId = $conf->global->PAYPAL_PAYMENT_MODE_FOR_PAYMENTS;
-											}
-											if ($paymentmethod == 'stripe') {
-												$paymentTypeId = $conf->global->STRIPE_PAYMENT_MODE_FOR_PAYMENTS;
-											}
-											if (empty($paymentTypeId)) {
-												//erics
-												if ($sepaMode) {
-													$paymentType = 'PRE';
-												} else {
-													$paymentType = $_SESSION["paymentType"];
-													if (empty($paymentType)) {
-														$paymentType = 'CB';
-													}
-												}
-												$paymentTypeId = dol_getIdFromCode($this->db, $paymentType, 'c_paiement', 'code', 'id', 1);
-											}
-
-											$currencyCodeType = $currency;
-
-											$ispostactionok = 1;
-
-											// Creation of payment line
-											// TODO LMR This must be move into the stripe server listening hooks public/stripe/ipn.php
-											include_once DOL_DOCUMENT_ROOT . '/compta/paiement/class/paiement.class.php';
-											$paiement = new Paiement($this->db);
-											$paiement->datepaye = $now;
-											$paiement->date = $now;
-											if ($currencyCodeType == $conf->currency) {
-												$paiement->amounts = [$this->id => $amounttopay];   // Array with all payments dispatching with invoice id
-											} else {
-												$paiement->multicurrency_amounts = [$this->id => $amounttopay];   // Array with all payments dispatching
-
-												$postactionmessages[] = 'Payment was done in a different currency than currency expected of company';
-												$ispostactionok = -1;
-												// Not yet supported, so error
+											$result =$this->setStatut($this::STATUS_PROCESSING);
```

Comment:
The fact that there is something "processing" must not be store into the invoice but into the llx_prelevement_bon (that must be created where there is the // TODO Create a prelevement_bon)
Because 1 invoice can have 2 request: 1 done, 1 pending, and a remain to pay not started. So no need to add a status on invoice. We already know there is some request in the pipe with the table llx_prelevement_request (where is sotre the invoice and the amount) linked to llx_prelevement_bon (where is stored the status).